packages/@react-aria/list/docs/useList.mdx

```diff
@@ -0,0 +1,385 @@
+hook and render an `<input>` element directly, but it's likely you'll have a `Checkbox` component in your component library that uses these hooks already.
+See the [useCheckbox docs](useCheckbox.html) for more information.
+
+```tsx example export=true render=false
+import {useListSelectionCheckbox} from '@react-aria/list';
+import {useToggleState} from '@react-stately/toggle';
+import {useCheckbox} from '@react-aria/checkbox';
+
+function ListCheckbox({ item, state }) {
+  let ref = useRef();
+  let { checkboxProps } = useListSelectionCheckbox({ key: item.key }, state);
+
+  let inputRef = useRef(null);
+  let { inputProps } = useCheckbox(
+    checkboxProps,
+    useToggleState(checkboxProps),
+    inputRef
+  );
+
+  return (
+    <span ref={ref}>
+      <input {...inputProps} />
+    </span>
+  );
+}
+```
+
+The following example shows how to enable multiple selection support using the List component we built above.
+It's as simple as setting the `selectionMode` prop to `"multiple"`.
+
+```tsx example
+<List aria-label="List with selection" selectionMode="multiple">
+  <Item key="1">Charizard</Item>
+  <Item key="2">Blastoise</Item>
+  <Item key="3">Venusaur</Item>
+  <Item key="4">Pikachu</Item>
+</List>
+```
+
+And that's it! We now have a fully interactive List component that can support keyboard navigation, single or multiple selection.
+In addition, it is fully accessible for screen readers and other assistive technology. See below for more
+examples of how to use the List component that we've built.
+
+## Usage
+
+### Dynamic collections
+
+So far, our examples have shown static collections, where the data is hard coded.
+Dynamic collections, as shown below, can be used when the List data comes from an external data source such as an API, or updates over time.
+In the example below, the rows are provided to the List via a render function.
+
+```tsx example export=true
+function ExampleList(props) {
+  let rows = [
+    {id: 1, name: 'Games'},
+    {id: 2, name: 'Program Files'},
+    {id: 3, name: 'bootmgr'},
+    {id: 4, name: 'log.txt'}
+  ];
+
+  return (
+    <List aria-label="Example dynamic collection List" items={rows} {...props}>
+      {item => <Item>{item.name}</Item>}
+    </List>
+  );
+}
+```
+
+### Single selection
+
+By default, `useListState` doesn't allow row selection but this can be enabled using the `selectionMode` prop. Use `defaultSelectedKeys` to provide a default set of selected rows.
+Note that the value of the selected keys must match the `key` prop of the row.
+
+The example below enables single selection mode, and uses `defaultSelectedKeys` to select the row with key equal to "2".
+A user can click on a different row to change the selection, or click on the same row again to deselect it entirely.
+
+```tsx example
+// Using the example above
+<ExampleList selectionMode="single" defaultSelectedKeys={[2]} />
```

Comment:
You'll want to add aria-labels to the examples below so they can be distinguished from one another